test/cli_run_device_test.go

```diff
@@ -160,139 +182,30 @@ func (suite *PouchRunDeviceSuite) TestRunDeviceReadBps(c *check.C) {
 	}
 
 	c.Assert(len(result[0].HostConfig.BlkioDeviceReadBps), check.Equals, 1)
-	c.Assert(result[0].HostConfig.BlkioDeviceReadBps[0].Path,
-		check.Equals, testDisk)
-	c.Assert(result[0].HostConfig.BlkioDeviceReadBps[0].Rate,
-		check.Equals, uint64(1048576))
-
-	// test if cgroup has record the real value
-	containerID := result[0].ID
-	{
-		path := fmt.Sprintf(
-			"/sys/fs/cgroup/blkio/default/%s/blkio.throttle.read_bps_device",
-			containerID)
-		checkFileContains(c, path, "1048576")
-	}
-}
-
-// TestRunDeviceWriteBps tests running container
-// with --device-write-bps flag.
-func (suite *PouchRunDeviceSuite) TestRunDeviceWriteBps(c *check.C) {
-	cname := "TestRunDeviceWriteBps"
-	testDisk, found := environment.FindDisk()
-	if !found {
-		c.Skip("fail to find available disk for blkio test")
-	}
-
-	res := command.PouchRun("run", "-d",
-		"--device-write-bps", testDisk+":1mb",
-		"--name", cname, busyboxImage, "sleep", "10000")
-	defer DelContainerForceMultyTime(c, cname)
-	res.Assert(c, icmd.Success)
-
-	// test if the value is in inspect result
-	res = command.PouchRun("inspect", cname)
-	res.Assert(c, icmd.Success)
-
-	result := []types.ContainerJSON{}
-	if err := json.Unmarshal([]byte(res.Stdout()), &result); err != nil {
-		c.Errorf("failed to decode inspect output: %v", err)
-	}
-
-	c.Assert(len(result[0].HostConfig.BlkioDeviceWriteBps),
-		check.Equals, 1)
-	c.Assert(result[0].HostConfig.BlkioDeviceWriteBps[0].Path,
-		check.Equals, testDisk)
-	c.Assert(result[0].HostConfig.BlkioDeviceWriteBps[0].Rate,
-		check.Equals, uint64(1048576))
-
-	// test if cgroup has record the real value
-	containerID := result[0].ID
-	{
-		path := fmt.Sprintf(
-			"/sys/fs/cgroup/blkio/default/%s/blkio.throttle.write_bps_device",
-			containerID)
-		checkFileContains(c, path, "1048576")
-	}
-}
-
-// TestRunDeviceReadIops tests running container with --device-read-iops flag.
-func (suite *PouchRunDeviceSuite) TestRunDeviceReadIops(c *check.C) {
-	cname := "TestRunDeviceReadIops"
-	testDisk, found := environment.FindDisk()
-	if !found {
-		c.Skip("fail to find available disk for blkio test")
-	}
+	c.Assert(result[0].HostConfig.BlkioDeviceReadBps[0].Path, check.Equals, testDisk)
```

Comment:
just to confirm, is there a need to confim the len of `result[0].HostConfig.BlkioDeviceReadBps` first?

Reply:
I think it is better to do so, since the struct definition is like below:
```
	// Limit write rate (bytes per second) to a device, in the form `[{"Path": "device_path", "Rate": rate}]`.
	//
	BlkioDeviceWriteBps []*ThrottleDevice `json:"BlkioDeviceWriteBps"`
```
Check the length first is to make sure the result is correct. @ZYecho

Reply:
but seems not check the len first here?